static/app/components/events/eventReplay/index.tsx

```diff
@@ -50,7 +50,7 @@ function EventReplayContent({
 
   const platform = group?.project.platform ?? 'other';
   const newOnboarding = organization.features.includes('session-replay-new-zero-state');
-  if (replayBackendPlatforms.includes(platform) && newOnboarding) {
+  if (replayBackendPlatforms.includes(platform) && newOnboarding && !replayId) {
```

Comment:
If we move `newOnboarding` and `!replayId` before the O(n) operation, it would be faster, since we won't have to iterate through the `replayBackendPlatforms` array if replayId is missing.

Reply:
Also, if `replayId` is missing, we don't need to do the same O(n) operation for line 52.